optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/factory/MoveIteratorFactory.java

```diff
@@ -49,7 +50,7 @@
      * @throws UnsupportedOperationException if only {@link #createRandomMoveIterator(ScoreDirector, Random)} is
      * supported
      */
-    Iterator<Move> createOriginalMoveIterator(ScoreDirector<Solution_> scoreDirector);
+    Iterator<Move_> createOriginalMoveIterator(ScoreDirector<Solution_> scoreDirector);
```

Comment:
Yes, that works too. I started with this approach and it's covered by the first two commits. Then I realized it's possible to declare the move type like this. What I consider nice about this is that when you declare
```java
class MyMoveFactory implements MoveIteratorFactory<MySolution, MyMove> {}
```
and then let your IDE generate stubs implementing the interface, you get
```java
public Iterator<MyMove> createOriginalMoveIterator(ScoreDirector<MySolution> scoreDirector) {}
```
instead of
```java
public Iterator<? extends Move<MySolution>> createOriginalMoveIterator(ScoreDirector<MySolution> scoreDirector) {}
```
and I like the former better. But I understand why you consider the additional type parameter a pain so I'll remove commits 3 and 4.